contrib/mesos/pkg/executor/executor.go

```diff
@@ -208,23 +210,74 @@ func (k *KubernetesExecutor) isDone() bool {
 	}
 }
 
+func (k *KubernetesExecutor) NodeInfo() (ni NodeInfo) {
+	k.nodeInfoLock.Lock()
+	defer k.nodeInfoLock.Unlock()
+
+	// clone k.nodeInfo
```

Comment:
you're right, was thinking about cloning api objects which NodeInfo is not.

On Tue, Sep 8, 2015 at 9:45 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/executor/executor.go
> <https://github.com/kubernetes/kubernetes/pull/13036#discussion_r38926443>
> :
>
> > @@ -208,23 +210,74 @@ func (k *KubernetesExecutor) isDone() bool {
> >  	}
> >  }
> >
> > +func (k *KubernetesExecutor) NodeInfo() (ni NodeInfo) {
> > +	k.nodeInfoLock.Lock()
> > +	defer k.nodeInfoLock.Unlock()
> > +
> > +	// clone k.nodeInfo
>
> Does that work with custom structs? I thoughts it uses pregenerated deep
> copy functions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13036/files#r38926443>.
>